calliope/backend/pyomo/interface.py

```diff
@@ -171,6 +171,107 @@ def rerun_pyomo_model(model_data, backend_model):
     return new_calliope_model
 
 
+def add_pyomo_constraint(
+    backend_model, constraint_name, constraint_sets, constraint_rule
+):
+    """
+    Functionality to add a constraint to the Pyomo backend, by pointing it to
+    a function which implements the mathematics
+
+    Parameters
+    ----------
+    constraint_name : str
+        Name for the pyomo.Constraint object
+    constraint_sets : list of str
+        Sets over which to implement the constraint.
+        The list of sets in the Pyomo model corresponds to model dimensions and
+        can be found in full by calling the backend interface method `get_all_model_attrs`
+    constraint_rule : function with args (backend_model, *args)
+        Function corresponding to the format expected by Pyomo,
+        i.e. arguments include the Pyomo backend model (`backend_model`)
+        and pointers to each set in `constraint_sets`.
+        Set arguments need to be in the same order as in constraint_sets.
+        The convention we follow is to name the index without the trailing `s`
+        E.g. if 'timesteps' is in constraint_sets, then 'timestep' should be one of the function arguments;
+        long `loc_techs` set names are abbreviated to `loc_tech`.
+        To see what parameters and variables you have to work with,
+        call the backend interface method `get_all_model_attrs`
+
+    Examples
```

Comment:
Same comments as in the user documnentation, in particular I'd go for a simpler example and make crystal clear that any `build` step will kill these changes.